test_project/app/api/endpoints/shop_api.py

```
@@ -90,6 +93,65 @@ async def get_categories(
+        )
+
+        await Employee.async_queryset.with_transaction(transaction).create(
+            department_id=department.id,
+            name=payload.get('employee_name', 'Will Rollback'),
+            email=payload.get('employee_email', 'rollback.employee@example.com'),
+            salary=payload.get('employee_salary', 60000),
+            hire_date=date.today(),
+        )
+
+        # Trigger a failure so we can test rollback behaviour
+        raise ValueError('Forcing rollback for transaction demo')
```

Ruff (0.14.7)

Abstract `raise` to an inner function

(TRY301)